packages/cli/src/ui/App.test.tsx

```diff
@@ -1119,4 +1119,253 @@ describe('App UI', () => {
+          handleFolderTrustSelect: vi.fn(),
+        }));
+
+      // Mock handleSlashCommand to track if /quit was called
+      const mockHandleSlashCommand = vi.fn();
+      const originalUseSlashCommandProcessor = await import(
+        './hooks/slashCommandProcessor.js'
+      );
+      const useSlashCommandProcessorSpy = vi
+        .spyOn(originalUseSlashCommandProcessor, 'useSlashCommandProcessor')
+        .mockReturnValue({
+          handleSlashCommand: mockHandleSlashCommand,
+          slashCommands: [],
+          pendingHistoryItems: [],
+          commandContext: {},
+          shellConfirmationRequest: null,
+        });
+
+      const { stdin, unmount } = render(
+        <App
+          config={mockConfig as unknown as ServerConfig}
+          settings={mockSettings}
+          version={mockVersion}
+        />,
+      );
+      currentUnmount = unmount;
+
+      // Simulate first Ctrl+C keypress
+      stdin.write('\x03'); // Ctrl+C character
+
+      // Wait a bit (but less than the timeout duration)
+      await new Promise((resolve) => setTimeout(resolve, 10));
+
+      // Simulate second Ctrl+C keypress
+      stdin.write('\x03'); // Ctrl+C character
+
+      // Wait for the component to process the input
+      await new Promise((resolve) => setTimeout(resolve, 10));
+
+      // Verify /quit command was called
+      expect(mockHandleSlashCommand).toHaveBeenCalledWith('/quit');
+
+      // Clean up mocks
+      useFolderTrustSpy.mockRestore();
+      useSlashCommandProcessorSpy.mockRestore();
+    });
+
+    it('should cancel pending tool calls when Ctrl+C is pressed during streaming', async () => {
+      // Mock useGeminiStream to simulate a responding state with pending tool calls
+      const mockCancelRequest = vi.fn();
+      vi.mocked(useGeminiStream).mockReturnValue({
+        streamingState: StreamingState.Responding,
+        submitQuery: vi.fn(),
+        initError: null,
+        pendingHistoryItems: [
+          {
+            type: 'tool_group',
+            tools: [
+              {
+                name: 'test_tool',
+                status: 'pending',
+                description: 'A test tool call',
+              },
+            ],
+          },
+        ],
+        thought: null,
+        cancelRequest: mockCancelRequest,
+      });
+
+      const { stdin, unmount } = render(
+        <App
+          config={mockConfig as unknown as ServerConfig}
+          settings={mockSettings}
+          version={mockVersion}
+        />,
+      );
+      currentUnmount = unmount;
+
+      // Simulate Ctrl+C keypress
+      stdin.write('\x03'); // Ctrl+C character
+
+      // Wait for the component to process the input
+      await new Promise((resolve) => setTimeout(resolve, 10));
+
+      // Verify cancelRequest was called (which should handle tool call cancellation)
+      expect(mockCancelRequest).toHaveBeenCalled();
+    });
+
+    it('should behave the same as Escape key for cancellation', async () => {
```

Comment:
remove this test or fix so it makes more sense. it is a bit like two tests after each other.